src/fsharp/IlxGen.fs

```diff
@@ -2440,24 +2440,27 @@ and GenConstant cenv cgbuf eenv (c, m, ty) sequel =
       | Some e ->
           GenExpr cenv cgbuf eenv SPSuppress e Continue
       | None ->
-          match c with
-          | Const.Bool b -> CG.EmitInstr cgbuf (pop 0) (Push [g.ilg.typ_Bool]) (mkLdcInt32 (if b then 1 else 0))
-          | Const.SByte i -> CG.EmitInstr cgbuf (pop 0) (Push [ilTy]) (mkLdcInt32 (int32 i))
-          | Const.Int16 i -> CG.EmitInstr cgbuf (pop 0) (Push [ilTy]) (mkLdcInt32 (int32 i))
-          | Const.Int32 i -> CG.EmitInstr cgbuf (pop 0) (Push [ilTy]) (mkLdcInt32 i)
-          | Const.Int64 i ->
-            // see https://github.com/Microsoft/visualfsharp/pull/3620
+          let emitLongConstant i =
```

Comment:
nit: We normally don't use the name `Long` to represent a 64-bit int. I suggest re-name this to `emitInt64Constant`.

Reply:
I had been inspired by Roslyn and gave it that name. It sounded a but uncanny to me as well, but it's now fixed.